### PR TITLE
Added Docker push postpublish script

### PR DESCRIPTION
**Pull request by rishivikram**

### CODE REVIEW

File: docker/images/ubuntu1404/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# newman_ubuntu1404
+
+This image runs newman 3.9.0 on node 4.3.0 on Ubuntu 14.04.2
```
Comment (kunagpal) [inline docker/images/ubuntu1404/README.md]: Remove the Newman version from this statement, mention the Node version as 8.

File: docker/images/ubuntu1404/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Build the image,
+
+```terminal
+docker build -t postman/newman_ubuntu1404 .
+```
+
+Or get it from [docker hub](https://registry.hub.docker.com/u/postman/newman_ubuntu1404/)
+
+```terminal
+docker pull postman/newman_ubuntu1404:3.9.0
```
Comment (kunagpal) [inline docker/images/ubuntu1404/README.md]: Same as above.

File: docker/scripts/run.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			docker push "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG"
+		else
+			echo "Collection not run successfully"
+		fi
+	else
+		echo "Image not built"
+	fi
+}
+
+NODE_V=$(node -v | grep -o "v.")
```
Comment (kunagpal) [inline docker/scripts/run.sh]: Travis provides an inbuilt environment variable for this, use that instead.

File: docker/scripts/run.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+IMAGES_BASE_PATH="./docker/images"
+if [ -n "$TRAVIS_TAG" ] && [ "$NODE_V" = "v8" ]; then
```
Comment (kunagpal) [inline docker/scripts/run.sh]: This will cause unstable tags to be built as well.

File: docker/scripts/run.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+function build_docker_image {
+    BASENAME=$(basename $1)
+    IMAGE_NAME="newman_$BASENAME"
+	docker build -t "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG" .
+	if docker images | grep -q "$DOCKER_ID_USER/$IMAGE_NAME"; then
+		echo "Image built"
+		if docker run -t "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG" run "https://www.getpostman.com/collections/8a0c9bc08f062d12dcda"; then
```
Comment (kunagpal) [inline docker/scripts/run.sh]: This will fail if the link ever dies, use a local collection file instead.

File: docker/scripts/run.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+			echo "Collection run successfully"
+			docker login -u "$DOCKER_ID_USER" -p "$DOCKER_ID_PASSWORD"
+			docker push "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG"
```
Comment (kunagpal) [inline docker/scripts/run.sh]: This needs the `latest` tag as well.

File: docker/scripts/run.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
+function build_docker_image {
+    BASENAME=$(basename $1)
+    IMAGE_NAME="newman_$BASENAME"
+	docker build -t "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG" .
```
Comment (kunagpal) [inline docker/scripts/run.sh]: Indentation is off.

File: docker/scripts/run.sh [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+		echo "Image built"
+		if docker run -t "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG" run "https://www.getpostman.com/collections/8a0c9bc08f062d12dcda"; then
+			echo "Collection run successfully"
+			docker login -u "$DOCKER_ID_USER" -p "$DOCKER_ID_PASSWORD"
+			docker push "$DOCKER_ID_USER/$IMAGE_NAME:$TRAVIS_TAG"
+		else
+			echo "Collection not run successfully"
+		fi
+	else
+		echo "Image not built"
```
Comment (kunagpal) [inline docker/scripts/run.sh]: Also include the image basename in the message.

File: docker/images/alpine33/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,40 @@
+FROM alpine:3.3
+MAINTAINER Postman Labs <help@getpostman.com>
+
+# Set node version
+# Can't upgrade to nodev8 because it isn't supported in alpine 3.3
```
Comment (kunagpal) [inline docker/images/alpine33/Dockerfile]: 👍

File: docker/images/alpine33/README.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
+# newman_alpine33
+
+This image runs newman 3.9.0 on node 4.3.0 on Alpine 3.3
```
Comment (kunagpal) [inline docker/images/alpine33/README.md]: Remove the Newman version, update the Node version to `v4.3.2`

File: docker/scripts/run.sh [hunk 11]
```diff
@@ -0,0 +1,31 @@
+function build_docker_image {
+    BASENAME=$(basename $1);
+    IMAGE_NAME="newman_$BASENAME";
+    TAG="$2";
+    docker build -t "$DOCKER_ID_USER/$IMAGE_NAME:$TAG" .
```
Comment (kunagpal) [inline docker/scripts/run.sh]: You'll have to add the latest tag here (-t can accept multiple space separated name:value parameters). See https://github.com/postmanlabs/newman-docker/blob/develop/scripts/docker/docker-build.sh#L13 for a working example.

File: docker/scripts/run.sh [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+IMAGES_BASE_PATH="./docker/images"
+EXTRACTED_TAG=$(echo $TRAVIS_TAG | grep -o "[0-9]\+.[0-9]\+.[0-9]\+")
+if [ -n "$TRAVIS_TAG" ] && [ "$TRAVIS_TAG" = "$EXTRACTED_TAG" ] && [ "$TRAVIS_NODE_VERSION" = "8" ]; then
+    for image in $IMAGES_BASE_PATH/*; do
+        if [ -d "${image}" ] && [ -f "${image}/Dockerfile" ]; then
+            cp "${image}/Dockerfile" .
+            build_docker_image ${image} ${TRAVIS_TAG};
+            build_docker_image ${image} "latest";
```
Comment (kunagpal) [inline docker/scripts/run.sh]: You don't have to build two separate images, supply multiple tags as explained above. 😓

File: docker/scripts/run.sh [hunk 13]
```diff
@@ -0,0 +1,31 @@
+function build_docker_image {
+    BASENAME=$(basename $1);
+    IMAGE_NAME="newman_$BASENAME";
+    TAG="$2";
+    docker build -t "$DOCKER_ID_USER/$IMAGE_NAME:$TAG" .
+    if docker images | grep -q "$DOCKER_ID_USER/$IMAGE_NAME"; then
+        echo "Image built"
+        if docker run -t "$DOCKER_ID_USER/$IMAGE_NAME:$TAG" run "sample-collection.json"; then
```
Comment (kunagpal) [inline docker/scripts/run.sh]: Don't you have to specify the complete path, as in `examples/sample-collection.json`?

Comment (rishivikram) [reply docker/scripts/run.sh]: In this case, I already mounted the whole `examples` directory in `etc/newman`, so no need. The collection runs correctly.

File: docker/scripts/run.sh (hunk 12 above)
Comment (rishivikram) [reply docker/scripts/run.sh]: Sorry, will correct it right away.

File: docker/scripts/run.sh (hunk 13 above)
Comment (rishivikram) [reply docker/scripts/run.sh]: I did the mounting work in the dockerfile.

File: test/system/repository.test.js [hunk 14]
```diff
@@ -73,19 +73,26 @@ describe('project repository', function () {
...
+                    expect(fs.existsSync('npm/' + scriptName + '.js') ||
+                        fs.existsSync('npm/' + scriptName + '.sh')).to.be.ok();
                 });
             });
 
             it('must have the hashbang defined', function () {
                 json.scripts && Object.keys(json.scripts).forEach(function (scriptName) {
-                    var fileContent = fs.readFileSync('npm/' + scriptName + '.js').toString();
-                    expect(/^#!\/(bin\/bash|usr\/bin\/env\snode)[\r\n][\W\w]*$/g.test(fileContent)).to.be.ok();
+                    fs.readFile('npm/' + scriptName + '.js', function (err, fileContent) {
```
Comment (kunagpal) [inline test/system/repository.test.js]: No. Check if the file is a shell script or js file using fs.stat first, and then perform extension specific checks.

File: test/system/repository.test.js [hunk 15]
```diff
@@ -73,19 +73,26 @@ describe('project repository', function () {
 
         describe('script definitions', function () {
             it('files must exist', function () {
-                var scriptRegex = /^node\snpm\/.+\.js$/;
-
+                var scriptRegexJS = /^node\snpm\/.+\.js$/,
+                    scriptRegexSH = /^\.\/npm\/.+\.sh$/;
```
Comment (kunagpal) [inline test/system/repository.test.js]: Have one cumulative regex testing both patterns.

File: docker/images/ubuntu1404/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+# Set locale
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+
+# Install node
+RUN npm install -g n;
+RUN n ${NODE_VERSION};
+
+ARG newman_version=default_value
```
Comment (kunagpal) [inline docker/images/ubuntu1404/Dockerfile]: ?

File: docker/images/alpine33/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+
+# Set locale
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US.UTF-8
+
+# Install node
+RUN apk add --update nodejs=${NODE_VERSION}-r1;
+
+ARG newman_version=default_value
```
Comment (kunagpal) [inline docker/images/alpine33/Dockerfile]: ?

File: docker/images/ubuntu1404/Dockerfile (hunk 16 above)
Comment (rishivikram) [reply docker/images/ubuntu1404/Dockerfile]: It needs a default value which would be used if $newman_version is not sent as a variable in the docker build command.

**Comment by guessi** [comment]
@rishivikram FYI,
maybe you would like to see my PR at another repository:
- [newman-docker - Pull Request 89](https://github.com/postmanlabs/newman-docker/pull/89)

File: docker/images/ubuntu1404/README.md [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+
+This image runs newman on node v8 on Ubuntu 14.04.2
+
+Build the image,
+
+```terminal
+docker build -t postman/newman_ubuntu1404 .
+```
+
+Or get it from [docker hub](https://registry.hub.docker.com/u/postman/newman_ubuntu1404/)
```
Comment (guessi) [inline docker/images/ubuntu1404/README.md]: please update the path on the Docker Hub
- https://hub.docker.com/r/postman/newman_ubuntu1404/

**Comment by liyinchigithub** [comment]
deprecation ！why？

**Comment by kunagpal** [comment]
@liyinchigithub This has now been replaced with https://https://hub.docker.com/r/postman/newman, which has all tags under one unified repository.

**Pull request merged**